ext/standard/string.c

```diff
@@ -4794,7 +4794,9 @@ PHP_FUNCTION(setlocale)
 				/* Remember if locale was changed */
 				size_t len = strlen(retval);
 
-				BG(locale_changed) = 1;
+				if (!zend_string_equals_literal(loc, "C")) {
```

Comment:
how about setlocale “c”?  maybe zend_string_equals_literal_ci here?

Reply:
nevermind, it should not valid.  I think this change is fine, after this change , we could also change https://github.com/php/php-src/blob/master/ext/pcre/php_pcre.c#L599